Expensify/App
# Update "Additional details" form in KYC wallet upgrade flow

### Details

Improves the "Additional details" step of the KYC Wallet Upgrade flow so it behaves like the VBA.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/188531

### Tests
1. Login with an account that has a SILVER wallet
2. Trigger the KYC flow by attempting to send money to another user via the FAB + button
3. Upload doc and video in the Onfido screen
4. Test the "Additional details" form on the next screen
5. Make sure submitting the form without any details will highlight which fields are required
6. Test that you cannot add an address with a PO Box
7. Test that server errors are presented above the "Save & continue" button and validation errors have a "Fix the errors" link
8. Close form and re-trigger. Verify that the user details are not cleared out and "remembered".

### QA Steps
Internal QA - perform tests above with a new account on the "Send Money" beta

### Tested On

- [x] Mobile Web - Onfido does not work on simulator so I manually skipped that step

### Screenshots


#### Web
![2021-12-13_18-54-33](https://user-images.githubusercontent.com/32969087/145935534-3b59a5f5-e3d7-423a-88a3-ecfc67434f41.png)
![2021-12-13_18-54-24](https://user-images.githubusercontent.com/32969087/145935541-5533fc1c-6e4f-4117-87ee-3fb6733181c0.png)
![2021-12-13_18-54-16](https://user-images.githubusercontent.com/32969087/145935545-50b7a104-5531-49ea-bf47-0bd8e9079dbf.png)

#### Mobile Web
![2021-12-14_08-23-30](https://user-images.githubusercontent.com/32969087/146057988-12995b17-f54e-46bb-b1b6-de9f4178007d.png)
![2021-12-14_08-08-54](https://user-images.githubusercontent.com/32969087/146057998-c3fd1dd0-6d05-4e7f-8fb1-7bcb4d0695d4.png)
![2021-12-14_08-08-33](https://user-images.githubusercontent.com/32969087/146058002-11f0ce0e-0b7e-49f7-9c50-7b23a1bf4a58.png)

#### Desktop
![2021-12-14_08-33-15](https://user-images.githubusercontent.com/32969087/146058896-7148ba51-cb23-4c56-a50f-295785efa551.png)
![2021-12-14_08-32-25](https://user-images.githubusercontent.com/32969087/146058902-64fa69e6-6ee4-4f08-b9cd-f617445ae511.png)

#### iOS
![2021-12-14_10-17-44](https://user-images.githubusercontent.com/32969087/146073134-ec8e0e1e-083c-4126-abb6-eb23000ae53e.png)

#### Android
![2021-12-14_10-22-39](https://user-images.githubusercontent.com/32969087/146073808-49228b4e-3574-4e9a-a1a6-5a2824c79a69.png)
![2021-12-14_10-22-31](https://user-images.githubusercontent.com/32969087/146073811-bcdf139f-eef5-4f73-969e-3d6f72ad993b.png)
![2021-12-14_10-22-15](https://user-images.githubusercontent.com/32969087/146073813-63bf9376-1b10-4356-8a18-b0256b4a37da.png)
